test/ERC20CosmosCoinAmountConversionTest.t.sol

```diff
@@ -0,0 +1,436 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.25 <0.9.0;
+
+// solhint-disable custom-errors,max-line-length
+import { Test } from "forge-std/Test.sol";
+import { ERC20CosmosCoinAmountConversion } from "../src/utils/ERC20CosmosCoinAmountConversion.sol";
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+// Discuss - Do we want to move mock contract to a mock folder?
```

Comment:
There are already some mocks, might make sense to keep mocks separate, yes.
Maybe we can also reuse the ERC20 mocks that exist there to some extent also.

Reply:
Yes, please move or reuse the other ERC20 mock if possible.